nixos/modules/services/backup/burp.nix

```diff
@@ -0,0 +1,653 @@
+    protocol = ${toString cfg.client.protocol}
+
+    ca_burp_ca = ${cfg.client.ca_burp_ca}
+    ca_csr_dir = ${cfg.client.ca_csr_dir}
+    ssl_cert_ca = ${cfg.client.ssl_cert_ca}
+    ssl_cert = ${cfg.client.ssl_cert}
+    ssl_key = ${cfg.client.ssl_key}
+    ssl_key_password = ${cfg.client.ssl_key_password}
+    ssl_peer_cn = ${cfg.client.ssl_peer_cn}
+
+    ${concatMapStringsSep "\n" (x: "include = " + x) cfg.client.includes}
+
+    ${concatMapStringsSep "\n" (x: "exclude = " + x) cfg.client.excludes}
+
+    nobackup = ${cfg.client.nobackup}
+
+    ${cfg.client.extraConfig}
+  '';
+
+  burp_server_conf = pkgs.writeText "burp-server.conf" ''
+    mode = server
+    pidfile = /run/burp/burp.server.pid
+
+    user = ${cfg.server.user}
+    group = ${cfg.server.group}
+
+    clientconfdir = ${cfg.server.clientconfdir}
+    ${concatMapStringsSep "\n" (x: "address = " + x) cfg.server.addresses}
+    port = ${toString cfg.server.port}
+    status_port = ${toString cfg.server.status_port}
+    directory = ${cfg.server.directory}
+    dedup_group = ${cfg.server.dedup_group}
+    protocol = ${toString cfg.server.protocol}
+    hardlinked_archive = ${toString cfg.server.hardlinked_archive}
+    working_dir_recovery_method = ${cfg.server.working_dir_recovery_method}
+    umask = ${toString cfg.server.umask}
+
+    client_can_delete = ${toString cfg.server.client_can_delete}
+    client_can_force_backup = ${toString cfg.server.client_can_force_backup}
+    client_can_list = ${toString cfg.server.client_can_list}
+    client_can_restore = ${toString cfg.server.client_can_restore}
+    client_can_verify = ${toString cfg.server.client_can_verify}
+
+    ca_conf = ${pkgs.writeText "CA.cnf" cfg.server.ca_conf_template}
+    ca_name = ${cfg.server.ca_name}
+    ca_server_name = ${cfg.server.ca_server_name}
+    ca_burp_ca = ${cfg.server.ca_burp_ca}
+
+    ca_crl_check = ${toString cfg.server.ca_crl_check}
+    ssl_cert_ca = ${cfg.server.ssl_cert_ca}
+    ssl_cert = ${cfg.server.ssl_cert}
+    ssl_key = ${cfg.server.ssl_key}
+    ssl_dhfile = ${cfg.server.ssl_dhfile}
+    ssl_key_password = ${cfg.server.ssl_key_password}
+    ssl_peer_cn = ${cfg.server.ssl_peer_cn}
+
+    ${concatMapStringsSep "\n" (x: "keep = " + toString x) cfg.server.keeps}
+
+    timer_script = ${cfg.server.timer_script}
+    ${concatMapStringsSep "\n" (x: "timer_arg = " + x) cfg.server.timer_args}
+
+    ${cfg.server.extraConfig}
+  '';
+
+  backup_client = pkgs.writeText "${cfg.client.cname}" ''
+    password = ${cfg.client.password}
+  '';
+
+in {
+  options = {
+    services.burp.client = {
+      enable = mkEnableOption "BURP client";
+
+      libDir = mkOption {
+        type = types.str;
+        default = "${libDir}/client";
+        description = ''
+          Location where client files are stored
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "burp";
+        description = ''
+          Run server as user
```

Comment:
Is this the client or the server config?

Reply:
The server but I'm not sure I understand what the question is getting at.